src/lib/pattern-utils.js

```
@@ -20,24 +20,247 @@ function matchUrl(url, searchPattern) {
+        case "trim":
+            return value.trim();
+        case "capitalize":
+            return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
+        case "title":
+            return value.replace(/(?:^|[\s\-_])([a-z])/gi, match => match.toUpperCase());
+        case "replace":
+            if (args.length < 2) throw new Error("replace requires 2 arguments");
+            return value.split(args[0]).join(args[1]);
+        case "limit":
+            if (args.length < 1) throw new Error("limit requires at least 1 argument");
+            const limitVal = parseInt(args[0], 10);
```

Biome (2.4.15)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)